snafu/benchmarks/nighthawk/nighthawk.py

```diff
@@ -0,0 +1,272 @@
+            latency_percentiles[percentile] += each_percentile['Value'] * 1000
+
+        return NighthawkStat(
+            requested_qps=data['RequestedQPS'],
+            actual_qps=data["ActualQPS"], 
+            throughput=data["ActualQPS"],
+            status_codes=data["RetCodes"],
+            p50_latency=latency_percentiles.get("50", None),
+            p75_latency=latency_percentiles.get("75", None),
+            p80_latency=latency_percentiles.get("80", None),
+            p90_latency=latency_percentiles.get("90", None),
+            p95_latency=latency_percentiles.get("95", None),
+            p99_latency=latency_percentiles.get("99", None),
+            p99_9_latency=latency_percentiles.get("99.9", None),
+            avg_latency=duration_histogram['Avg'] * 1000,
+            timestamp=data['StartTime'],
+            bytes_in=float(data['BytesReceived']),
+            bytes_out=float(data['BytesSent'])
+        )
+
+    def _run_nighthawk(self):
+        """
+        Method to execute nighthawk command.
+        """
+
+        cmd = (
+            "nighthawk_client --concurrency {0} --duration {1} --connections {2} "
+            "--max-requests-per-connection {3} --rps {4} --output-format fortio {5} > nighthawk.json"
+        ).format(self.config.concurrency, self.config.duration, self.config.connections, 
+                self.config.max_requests_per_connection, self.config.rps, self.config.url)
+        self.logger.info(cmd)
+        p = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        return p.stdout.strip().decode("utf-8"), p.stderr.strip().decode("utf-8"), p.returncode
+    
+    def _json_payload(self, data, sample):
+        """
+        Method to return the final nighthawk output of a sample.
+        """
+
+        data.iteration = sample
+        return NighthawkOut(
+            workload="nighthawk",
+            uuid=self.config.uuid,
+            user=self.config.user,
+            cluster_name=os.getenv("clustername", "mycluster"),
+            duration=self.config.duration,
+            targets=[self.config.url],
+            execution_results=data,
+            concurrency=self.config.concurrency,
+            connections=self.config.connections,
+            max_requests_per_connection=self.config.max_requests_per_connection,
+            hostname=socket.gethostname()
+        )
+
+    def collect(self) -> Iterable[BenchmarkResult]:
+        """
+        Run nighthawk benchmark ``self.config.sample`` number of times.
+
+        Returns immediately if a sample fails. Will attempt to Nighthawk run for each sample.
+        """
+
+        cmd = shlex.split(f"nighthawk_client --concurrency {self.config.concurrency} --duration {self.config.duration} --connections {self.config.connections} --max-requests-per-connection {self.config.max_requests_per_connection} --rps {self.config.rps} --output-format fortio {self.config.url} > nighthawk.json")
```

Comment:
As discussed on our call to just nix this line as it isn't used.

Reply:
Great work @vishnuchalla ! 🚀

Reply:
Removed that redundant line

Reply:
Also flattened the ES doc as discussed over the call.